### PR TITLE
useResponsive hook and modules for functionality

### PR DESCRIPTION
**Pull request by hi2dmitri**
Modularized code: Created useResponsiveHook to move logic controlling responsiveness to separate function, removed ScatterPlotBody component, created modules to store separate functionality responsible for determining scales and voronoi, created Circles and VoronoiCell components.

### CODE REVIEW

File: src/functionality/xScale.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as d3 from 'd3';
+import { Margin, Data, ScaleFunc, AccessorFunc, Domain } from "../../types"
+
+
+
+export function xScaleDef (data: Data[], xDataType: 'date' | 'number', xAccessor: AccessorFunc, margin: Margin, width: number) {
```
Comment (eric-mulhern) [inline src/functionality/xScale.tsx]: order of shared arguments should be the same in xScaleDef and yScaleDef

File: src/functionality/xScale.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      xMin = d3.extent(data, xAccessor)[0]
+      xMax = d3.extent(data, xAccessor)[1]
+      xScale = d3
+        .scaleLinear()
+        .domain([xMin ?? 0, xMax ?? 0])
+        .range([0, width - margin.right - margin.left])
+        .nice()
+      break
+    case "date":
+      xMin = d3.extent(data, xAccessor)[0]
```
Comment (eric-mulhern) [inline src/functionality/xScale.tsx]: invoking d3.extent twice is redundant. Use d3.min and d3.max instead (see AreaChartBody for example)

File: src/functionality/yScale.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import * as d3 from 'd3';
+import { Margin, Data, Domain, AccessorFunc } from "../../types"
+
+
+export function yScaleDef (data: Data[], margin: Margin, height: number, yAccessor: AccessorFunc) {
+
+  let yScale: d3.ScaleLinear<number, number, never>, yMin: Domain, yMax: Domain
+  yMin = d3.extent(data, yAccessor)[0]
+  yMax = d3.extent(data, yAccessor)[1]
```
Comment (eric-mulhern) [inline src/functionality/yScale.tsx]: invoking d3.extent twice is redundant. Use d3.min and d3.max instead (see AreaChartBody for example)

File: src/functionality/yScale.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  yScale = d3
+    .scaleLinear()
```
Comment (eric-mulhern) [inline src/functionality/yScale.tsx:10]: For some charts we will need to add more scaling options (for example scaleLog and scalePow for line chart), both in x scale and y scale. Doing so will be simple enough, we can just add a switch statement like in xScale. However, with the scale functionality abstracted, that means we're also giving that functionality to other charts that don't need it (like area and bar). We can either ignore this, or we can create multiple xScale and yScale functions that extend each other (e.g. area and bar use xScale & yScale, and line and scatter use xScaleWithOptions & yScaleWithOptions; functions that extend xScale & yScale by adding more choices).

File: types.ts [hunk 5]
```diff
@@ -152,3 +152,20 @@ export interface CircleProps {
 
 export type ColorScale = d3.ScaleOrdinal<string, string, never>
 
+export type ScaleFunc =
+  | d3.ScaleLinear<number, number, never>
+  | d3.ScaleTime<number, number, never>
+
+export type AccessorFunc = (d: any) => number | Date;
```
Comment (eric-mulhern) [inline types.ts:173]: do we want to make d of type Data?

**Review by eric-mulhern — COMMENTED** [review]
A few minor details to clean up, otherwise looks really good!

File: src/functionality/yScale.tsx (hunk 4 above)
Comment (hi2dmitri) [reply src/functionality/yScale.tsx:10]: This is correct comment, and functionality should be extended as we restructure other charts

File: src/functionality/yScale.tsx (hunk 3 above)
Comment (hi2dmitri) [reply src/functionality/yScale.tsx]: Even better solution would be destructuring like so:

const [yMin, yMax] = d3.extent(data, yAccessor)
d3.min and d3.max still requires invocation of two methods

File: types.ts (hunk 5 above)
Comment (hi2dmitri) [reply types.ts:173]: Only Date or number, same one applies to x and y accessors and x may be date or number while y is only number

**Pull request merged**